aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/stash/StashTest.java

```diff
@@ -69,4 +70,26 @@
         assertEquals("{}", StashManager.stashesOf(b).toString());
     }
 
+    @Issue("JENKINS-31086")
+    @Test public void testDefaultExcludes() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                        "  writeFile file: 'subdir/.gitignore', text: 'whatever'\n" +
+                        "  writeFile file: 'subdir/other', text: 'more'\n" +
+                        "  dir('subdir') {stash name:'whatever', useDefaultExcludes: false}\n" +
+                        "}\n" +
+                        "node {\n" +
+                        "  dir('elsewhere') {\n" +
+                        "    unstash 'whatever'\n" +
+                        "    echo \"got .gitignore: ${readFile '.gitignore'} other: ${readFile 'other'}\"\n" +
+                        "  }\n" +
+                        "  stash name: 'from-top', includes: 'elsewhere/'\n" +
+                        "}", true));
+        WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+        assertEquals("{from-top={elsewhere/other=more}, whatever={.gitignore=whatever, other=more}}", StashManager.stashesOf(b).toString());
```

Comment:
FYI, to get this to work, I did need to create a second file so that the `stash` with implicit `useDefaultExcludes` doesn't error out due to no files to stash.